rules/group.go

```diff
@@ -370,6 +372,28 @@ func (g *Group) setEvaluationTime(dur time.Duration) {
 	g.evaluationTime = dur
 }
 
+// GetRuleEvaluationTimeSum returns the sum of the time in seconds it took to evaluate each rule in the group.
+func (g *Group) GetRuleEvaluationTimeSum() time.Duration {
+	g.mtx.Lock()
+	defer g.mtx.Unlock()
+	return g.evaluationRuleTimeSum
+}
+
+// updateRuleEvaluationTimeSum updates the sum of the time in seconds it took to evaluate each rule in the group.
+// It collects the times from the rules themselves.
+func (g *Group) updateRuleEvaluationTimeSum() {
+	var sum time.Duration
+	for _, rule := range g.rules {
```

Comment:
I did a quick scan and I think this is correct, in the sense that `g.rules` can be accessed without the mutex protection but can you please verify you haven't missed something?

Reply:
`Get/SetEvaluationDuration` maps to an atomic in both recording and alerting rules. Additionally, `SetEvaluationDuration` is only called during the group's eval function

The only case it could be weird is if evaluations are being missed. Evaluations overlap in that case, right?

Reply:
Note that it wouldn't crash. You'd just have a total rule duration from two different evaluations. A group consistently missing evaluation is something to fix pretty quickly in any case, not sure if it's a big issue

Reply:
Is not possible to be in that state, if we hit the interval and haven't finished we just skip the evaluation: https://github.com/prometheus/prometheus/blob/main/rules/group.go#L272C4-L272C10